core/shared/src/main/scala/scalaz/zio/UIO.scala

```diff
@@ -0,0 +1,285 @@
+   */
+  final def descriptor: UIO[Fiber.Descriptor] = ZIO.descriptor
+
+  /** See [[scalaz.zio.ZIO.descriptorWith]]
+   */
+  final def descriptorWith[A](f: Fiber.Descriptor => UIO[A]): UIO[A] =
+    ZIO.descriptorWith(f)
+
+  /** See [[scalaz.zio.ZIO.die]]
+   */
+  final def die(t: Throwable): UIO[Nothing] = ZIO.die(t)
+
+  /** See [[scalaz.zio.ZIO.dieMessage]]
+   */
+  final def dieMessage(message: String): UIO[Nothing] = ZIO.dieMessage(message)
+
+  /** See [[scalaz.zio.ZIO.done]]
+   */
+  final def done[A](r: Exit[Nothing, A]): UIO[A] = ZIO.done(r)
+
+  /** See [[scalaz.zio.ZIO.effectTotal]]
+   */
+  final def effectTotal[A](effect: => A): UIO[A] = ZIO.effectTotal(effect)
+
+  /** See [[scalaz.zio.ZIO.effectTotalWith]]
+   */
+  final def effectTotalWith[A](effect: Platform => A): UIO[A] = ZIO.effectTotalWith(effect)
+
+  /** See [[scalaz.zio.ZIO.effectAsync]]
+   */
+  final def effectAsync[A](register: (UIO[A] => Unit) => Unit): UIO[A] =
+    ZIO.effectAsync(register)
+
+  /** See [[scalaz.zio.ZIO.effectAsyncMaybe]]
+   */
+  final def effectAsyncMaybe[A](register: (UIO[A] => Unit) => Option[UIO[A]]): UIO[A] =
+    ZIO.effectAsyncMaybe(register)
+
+  /** See [[scalaz.zio.ZIO.effectAsyncM]]
+   */
+  final def effectAsyncM[A](register: (UIO[A] => Unit) => UIO[_]): UIO[A] =
+    ZIO.effectAsyncM(register)
+
+  /** See [[scalaz.zio.ZIO.effectAsyncInterrupt]]
+   */
+  final def effectAsyncInterrupt[A](register: (UIO[A] => Unit) => Either[Canceler, UIO[A]]): UIO[A] =
+    ZIO.effectAsyncInterrupt(register)
+
+  /** See [[scalaz.zio.ZIO.flatten]]
+   */
+  final def flatten[A](zio: UIO[UIO[A]]): UIO[A] =
+    ZIO.flatten(zio)
+
+  /** See [[scalaz.zio.ZIO.foldLeft]]
+   */
+  final def foldLeft[S, A](in: Iterable[A])(zero: S)(f: (S, A) => UIO[S]): UIO[S] =
+    ZIO.foldLeft(in)(zero)(f)
+
+  /** See [[scalaz.zio.ZIO.forkAll]]
+   */
+  final def forkAll[A](as: Iterable[UIO[A]]): UIO[Fiber[Nothing, List[A]]] =
+    ZIO.forkAll(as)
+
+  /** See [[scalaz.zio.ZIO.forkAll_]]
+   */
+  final def forkAll_[A](as: Iterable[UIO[A]]): UIO[Unit] =
+    ZIO.forkAll_(as)
+
+  /** See [[scalaz.zio.ZIO.fromFunction]]
+   */
+  final def fromFunction[A](f: Any => A): UIO[A] =
+    ZIO.fromFunction(f)
+
+  /** See [[scalaz.zio.ZIO.fromFunctionM]]
+   */
+  final def fromFunctionM[A](f: Any => UIO[A]): UIO[A] =
+    ZIO.fromFunctionM(f)
+
+  /** See [[scalaz.zio.ZIO.fromEither]]
+   */
+  final def fromEither[A](v: => Either[Nothing, A]): UIO[A] =
+    ZIO.fromEither(v)
+
+  /** See [[scalaz.zio.ZIO.fromFiber]]
+   */
+  final def fromFiber[A](fiber: => Fiber[Nothing, A]): UIO[A] =
+    ZIO.fromFiber(fiber)
+
+  /** See [[scalaz.zio.ZIO.fromFiberM]]
+   */
+  final def fromFiberM[A](fiber: UIO[Fiber[Nothing, A]]): UIO[A] =
+    ZIO.fromFiberM(fiber)
+
+  /** See [[scalaz.zio.ZIO.halt]]
+   */
+  final def halt(cause: Cause[Nothing]): UIO[Nothing] = ZIO.halt(cause)
+
+  /** See [[scalaz.zio.ZIO.identity]]
+   */
+  final def identity: UIO[Any] = ZIO.identity
```

Comment:
This function only makes sense when `R` != `Any`, so should be deleted from `UIO`.